src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,857 @@
+							}
+
+							RemoveNodeAndDescendantsFromView(removedNode);
+							Insert(removedNodeIndex, targetNode);
+
+							if (IsContentMode)
+							{
+								m_itemToNodeMap.Remove(removedNode.Content);
+								m_itemToNodeMap.Add(targetNode.Content, targetNode);
+							}
+						}
+
+						break;
+					}
+			}
+		}
+
+		private void SelectedNodeChildrenChanged(TreeViewNode sender, object args)
+		{
+			var vectorChangedArgs = (IVectorChangedEventArgs)args;
+			CollectionChange collectionChange = vectorChangedArgs.CollectionChange;
+			var index = (int)vectorChangedArgs.Index; // Cast to int to be able to index
+			var changingChildrenNode = (TreeViewNode)sender;
+
+			switch (collectionChange)
+			{
+				case (CollectionChange.ItemInserted):
+					{
+						var newNode = changingChildrenNode.Children[index];
+						UpdateNodeSelection(newNode, NodeSelectionState(changingChildrenNode));
+						break;
+					}
+
+				case (CollectionChange.ItemChanged):
+					{
+						var newNode = changingChildrenNode.Children[index];
+						UpdateNodeSelection(newNode, NodeSelectionState(changingChildrenNode));
+
+						var selectedNodes = m_selectedNodes;
+						for (var i = 0; i < selectedNodes.Count; i++)
+						{
+							var selectNode = selectedNodes[i];
+							var ancestorNode = selectNode.Parent;
+							while (ancestorNode != null && ancestorNode.Parent != null)
+							{
+								ancestorNode = ancestorNode.Parent;
+							}
+
+							if (ancestorNode != m_originNode)
+							{
+								selectedNodes.RemoveAtCore(i);
+								//TODO
+								//m_selectedNodeChildrenChangedEventTokenVector.erase(m_selectedNodeChildrenChangedEventTokenVector.begin() + i);
+							}
+						}
+						break;
+					}
+
+				case (CollectionChange.ItemRemoved):
+				case (CollectionChange.Reset):
+					{
+						//This checks if there are still children, then re-evaluates parents selection based on current state of remaining children
+						//If a node has 2 children selected, and 1 unselected, and the unselected is removed, we then change the parent node to selected.
+						//If the last child is removed, we preserve the current selection state of the parent, and this code need not execute.
+						if (changingChildrenNode.Children.Count > 0)
+						{
+							var firstChildNode = changingChildrenNode.Children[0];
+							UpdateSelectionStateOfAncestors(firstChildNode);
+						}
+
+						var selectedNodes = m_selectedNodes;
+						for (int i = 0; i < selectedNodes.Count; i++)
+						{
+							var selectNode = selectedNodes[i];
+							var ancestorNode = selectNode.Parent;
+							while (ancestorNode != null && ancestorNode.Parent != null)
+							{
+								ancestorNode = ancestorNode.Parent;
+							}
+
+							if (ancestorNode != m_originNode)
+							{
+								selectedNodes.RemoveAtCore(i);
+								//TODO
+								//m_selectedNodeChildrenChangedEventTokenVector.erase(m_selectedNodeChildrenChangedEventTokenVector.begin() + i);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)